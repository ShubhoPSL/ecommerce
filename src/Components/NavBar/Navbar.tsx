import React from 'react'
import List from '../List/List'
import { navList } from '../../Data API/NavList'
import { Link } from 'react-router-dom'

function Navbar() {
    return (
        <>
        <nav className='main'>
            <div className="brand-div">
                <h2 className="brand-heading">
                    <Link to="/">
                        <h1 className="brand">ShopKart</h1>
                    </Link>
                </h2>
            </div>
<div className="menu">
            <ul className='list'>
                {navList.map((item)=>(
                    <List data={item} />
                ))}
            </ul>
            </div>
        </nav >
    </>
    )
}

export default Navbar