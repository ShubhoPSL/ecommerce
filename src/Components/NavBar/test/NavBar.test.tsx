import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { RecoilRoot } from "recoil";
import Navbar from "../Navbar";

jest.mock("../../List/List");

describe("Navbar is being tested", () => {
  const NewNavBar = () => {
    return (
      <RecoilRoot>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </RecoilRoot>
    );
  };
  test("should have text ShopKart", () => {
    render(<NewNavBar />);
    expect(screen.getByText("ShopKart")).toBeInTheDocument();
  });
  test("should have List item with text Home", () => {
    render(<NewNavBar />);
    expect(screen.getByText("List - Home")).toBeInTheDocument();
  });
  test("should have List item with text Products", () => {
    render(<NewNavBar />);
    expect(screen.getByText("List - Products")).toBeInTheDocument();
  });
  test("should have List item with text Contact Us", () => {
    render(<NewNavBar />);
    expect(screen.getByText("List - Contact Us")).toBeInTheDocument();
  });
  test("should match with snapshot of Navbar module", () => {
    const tree = renderer.create(<NewNavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
