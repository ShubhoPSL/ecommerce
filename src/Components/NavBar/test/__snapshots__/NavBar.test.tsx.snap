// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar is being tested should match with snapshot of Navbar module 1`] = `
<nav
  className="main"
>
  <div
    className="brand-div"
  >
    <h2
      className="brand-heading"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="brand"
        >
          ShopKart
        </h1>
      </a>
    </h2>
  </div>
  <div
    className="menu"
  >
    <ul
      className="list"
    >
      <div>
        List - 
        Home
      </div>
      <div>
        List - 
        Products
      </div>
      <div>
        List - 
        Contact Us
      </div>
      <div>
        List - 
        Cart
      </div>
    </ul>
  </div>
</nav>
`;
